From 6ce59f53bb1645eeefa66b4bd1aaab8401f07c0e Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Fri, 19 May 2023 16:34:12 +0200
Subject: [PATCH 0685/1695] plat-stm32mp2: conf: enable CRYP for stm32mp2
 platforms

Default enables CRYP and crypto framework for stm32mp2 platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2aa2d80254fe62df857fcabadd7ac9a27567389f plat-stm32mp2: conf: enable CRYP for stm32mp2 platforms]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/312663
Change-Id: Iaf5657cf910a2b4ada210f02ad3b3303188e7ef2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/363044
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 3c012913b..66de0910a 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -123,6 +123,7 @@ endif
 
 CFG_STM32_BSEC3 ?= y
 CFG_STM32_BSEC_WRITE ?= y
+CFG_STM32_CRYP ?= y
 CFG_STM32_EXTI ?= y
 CFG_STM32_FMC ?= y
 CFG_STM32_GPIO ?= y
@@ -182,7 +183,8 @@ override CFG_STM32_CRYP := n
 endif # cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES
 
 # If any crypto driver is enabled, enable the crypto-framework layer
-ifeq ($(call cfg-one-enabled, CFG_STM32_HASH CFG_STM32_PKA CFG_STM32_SAES),y)
+ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_HASH CFG_STM32_PKA \
+			      CFG_STM32_SAES ),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
-- 
2.47.2

